templates/client-daemonset.yaml

```diff
@@ -205,11 +205,18 @@ spec:
                 {{- if (and .Values.global.gossipEncryption.secretName .Values.global.gossipEncryption.secretKey) }}
                 -encrypt="${GOSSIP_KEY}" \
                 {{- end }}
+                {{- if or .Values.client.join (and .Values.externalServers.enabled .Values.externalServers.hosts) }}
                 {{- if .Values.client.join }}
                 {{- range $value := .Values.client.join }}
                 -retry-join={{ quote $value }} \
                 {{- end }}
                 {{- else }}
+                {{- $serfPort := (.Values.externalServers.serfLANPort | int) -}}
+                {{- range $host := .Values.externalServers.hosts }}
+                -retry-join={{ printf "%s:%d" $host $serfPort | quote}} \
```

Comment:
Echoing the comment I left (quite late) on the other PR: Does this still work if someone set a provider? Or a dns name? I think the docs also mention being able to set a go-sockaddr template.

Reply:
That's a good catch. I don't think this will work. I can change it to default the `serfLANPort` to null and only set it if it's provided in this template. I need to look into go-sockaddr.

Reply:
Nice catch. go-sockaddr templates look like `"{{ GetPrivateIP }}"` so as long as it's being echoed as a string in the case of serfPort being null you should be fine.

Reply:
There's also the (likely quite rare) case of someone having different ports for different servers.

Reply:
In that case they will only be able to set a single server but that's okay because once you reach one server it tells you the locations of the other servers.

Reply:
It'll be a confusing experience though- if they're trying to set multiple servers with different ports but can only set one port, which to choose? We'll also have drifted from Consul's definition of `retry-join`.

This awkwardness is why I recommended that we don't address the `client.join` switch and deprecation now. It means we'll need the port stripping in `consul-k8s`, but it gives us time to get this right.

Reply:
Could that confusion be addressed in a comment? The drift here is intentional because we're no longer shadowing the retry-join configuration. These hosts are used for two purposes/protocols, and so it made more sense to split the port out.

Reply:
There's also the option to not reuse `client.join` here and instead ask for a single server address and https port for this ACLs work. It potentially duplicates a config value, but doesn't run into any of the complications of trying to support all of the `retry-join` forms for clients while restricting what can be set for the acl connection.

That solves this problem without requiring special cases or port stripping in consul-k8s. Very tidy.

Reply:
> These hosts are used for two purposes/protocols

To me that means we should be asking for different configuration settings. We're overly complicating things trying to reuse this value.

Reply:
For posterity, we've decided to treat retry-join hosts and https hosts as separate properties.